apps/src/blockly/addons/cdoSpritePointer.ts

```diff
@@ -68,53 +75,72 @@ export function updatePointerBlockImage(
  *  should have a connection to a block of type gamelab_allSpritesWithAnimation, or we will return an empty string.
  * @returns Image url or empty string if the image source block does not have an image at the given index.
  */
-function getImageUrlFromImageSource(imageSourceBlock, imageIndexOnSource) {
+function getImageUrlFromImageSource(
+  imageSourceBlock: ExtendedBlockSvg,
+  imageIndexOnSource: number
+) {
   const targetConnection =
     imageSourceBlock.inputList[imageIndexOnSource]?.connection
       ?.targetConnection;
   // We only want to get the image from a connection block that is not an insertion marker.
   // If the block is an insertion marker that means the block is being dragged
   // and is not yet connected to the image source block.
+  const sourceBlock = targetConnection?.getSourceBlock();
```

Comment:
`sourceBlock_` is private, so we use `getSourceBlock()` instead (which returns `sourceBlock_`)